src/Compilers/Core/MSBuildTask/Microsoft.Build.Tasks.CodeAnalysis.csproj

```diff
@@ -60,5 +60,45 @@
   <ItemGroup>
     <InternalsVisibleTo Include="Microsoft.Build.Tasks.CodeAnalysis.UnitTests" />
   </ItemGroup>
+
+  <!-- Generate a .targets file that sets $(CompilerApiVersion) to the current Roslyn version -->
+  <PropertyGroup>
+    <CompilerVersionTargetsFileIntermediatePath>$(IntermediateOutputPath)Microsoft.Managed.Core.CurrentVersions.targets</CompilerVersionTargetsFileIntermediatePath>
+  </PropertyGroup>
+
+  <Target Name="GenerateCompilerVersionTargets"
+          BeforeTargets="AssignTargetPaths"
+          DependsOnTargets="GenerateCompilerVersionTargetsFile">
+    <ItemGroup>
+      <Content Include="$(CompilerVersionTargetsFileIntermediatePath)">
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+        <Pack>true</Pack>
+        <BuildAction>None</BuildAction>
+        <PackageCopyToOutput>true</PackageCopyToOutput>
+        <PackagePath>contentFiles\any\any</PackagePath>
+      </Content>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateCompilerVersionTargetsFile"
+          Inputs="$(MSBuildThisFileFullPath)"
+          Outputs="$(CompilerVersionTargetsFileIntermediatePath)">
+
+    <PropertyGroup>
+      <CompilerApiVersion>$([System.Version]::Parse($(VersionPrefix)).Major).$([System.Version]::Parse($(VersionPrefix)).Minor)</CompilerApiVersion>
+      <CompilerVersionTargetsFileContent>
+<![CDATA[<Project>
+  <PropertyGroup>
+    <CompilerApiVersion>roslyn$(CompilerApiVersion)</CompilerApiVersion>
```

Comment:
NOTE: an alternative, simpler approach could be to straight up hard-code `roslyn4.0` directly into `Microsoft.Managed.Core.targets`. And then rely on the test that I added to break whenever we bump the `Major/MinorVersion` properties in the repo - reminding the person who did the bumping to update the hard-coded version.

I'm open to either approach. I'll let the owners decide which is more preferable (or suggest another alternative).

Reply:
That's the exact approach I was concerned about, so I'm definitely in favor of doing it this way.